protocols/imap/src/main/java/org/apache/james/imap/decode/parser/SearchCommandParser.java

```diff
@@ -904,7 +904,7 @@ public SearchKey decode(ImapSession session, ImapRequestLineReader request) thro
 
     private ImapMessage unsupportedCharset(Tag tag, ImapCommand command) {
         final ResponseCode badCharset = StatusResponse.ResponseCode.badCharset();
-        return statusResponseFactory.taggedNo(tag, command, HumanReadableText.BAD_CHARSET, badCharset);
+        return getStatusResponseFactory().taggedNo(tag, command, HumanReadableText.BAD_CHARSET, badCharset);
```

Comment:
you could directly provide the `taggedNo` method in super class to avoid providing a getter